File: src/tutorials/code/node/interactive/4.js
function performRobotHandShake (robot) {
    return new Promise((resolve, reject) => {
        robot.handshake(err => {
            if (err) {
                reject(err);
            }
            resolve(robot);
        });
    });
}

function setupRobotEvents (robot) {
    function setupRobotEvents (robot) {
    robot.on('report', report => {
        const mouse = rjs.getMousePos();
        if (report.joystick.length > 0) {
            if (!isNaN(report.joystick[0].coordMean.x) && !isNaN(report.joystick[0].coordMean.y)) {
                rjs.moveMouse(
                    Math.round(mouse.x + 300 * report.joystick[0].coordMean.x),
                    Math.round(mouse.y + 300 * report.joystick[0].coordMean.y)
                );
            }
        }
    });
    robot.on('error', err => {
        throw new Error('There was an error in the Interactive connection', err);
    });
}

File: src/tutorials/code/java/interactive/4.java
package pro.beam.interactive.example;

import pro.beam.api.BeamAPI;
import pro.beam.interactive.net.packet.Protocol;
import pro.beam.interactive.robot.RobotBuilder;

import java.awt.*;

import java.util.concurrent.ExecutionException;

public class Main {

    public static void main(String[] args) throws AWTException {
        BeamAPI beam = new BeamAPI("AUTH_TOKEN");
        Robot controller = new Robot();
        BeamUser user = beam.use(UsersService.class).getCurrent().get();
        try {
            pro.beam.interactive.robot.Robot robot = new RobotBuilder()
                .channel(user.channel)
                .build(beam, false)
                .get();

            robot.on(Protocol.Report.class, report -> {
                // If we have any joysticks in the report
                if (report.getJoystickCount() > 0) {
                    // Get the coordMean from the joystick
                    Protocol.Coordinate coordMean = report.getJoystick(0).getCoordMean();
                    Point mousePosition = MouseInfo.getPointerInfo().getLocation();

                    // Apply it to the current mouse position, if its values are not NaN
                    if (!Double.isNaN(coordMean.getX()) && !Double.isNaN(coordMean.getY())) {
                        controller.mouseMove(
                            ((int) (mousePosition.getX() + 300 * coordMean.getX())),
                            ((int) (mousePosition.getY() + 300 * coordMean.getY()))
                        );
                    }
                }
            });

        } catch (InterruptedException | ExecutionException e) {
            e.printStackTrace();
        }
    }
}
